File: csharp/resistor-color/ResistorColor.cs
public static class ResistorColor
{
    private static string[] _colorBands = [
        "black", "brown", "red", "orange", "yellow",
        "green", "blue", "violet", "grey", "white"
    ];

    public static int ColorCode(string color)
        => _colorBands.IndexOf(color);  // IndexOf searches the array

    public static string[] Colors()
        => _colorBands;
}

File: fsharp/resistor-color/ResistorColor.fs
module ResistorColor

let colors: string list = [
    "black"; "brown"; "red"; "orange"; "yellow";
    "green"; "blue"; "violet"; "grey"; "white"
]

let colorCode (color: string): int =
    match List.tryFindIndex (fun x -> x = color) colors with
    | Some i -> i
    | None -> -1
